### ISSUE TITLE
When changing products, value_products autoupdates must be invoked (APIError: Line 1 has been deleted or never existed)

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#653]
_Imported from trac ticket 653.
Created by **kbower**
Opened in trac: Feb 4, 2011, 5:23:05 PM
Last modified in trac: Feb 4, 2011, 9:08:44 PM_
***
Wayne:
```
A fatal error was encountered when communicating with the server.

Error message:
APIError: Line 1 has been deleted or never existed (do not supply lineid when attempting to add new line)
```
orderlines.product must be added for process_value_products()

**Comment by kbower** [comment]
_**kbower** trac comment on Feb 4, 2011, 5:24:48 PM:_

Add modernscroll
Add MORG6
 (package explosion includes end table)
Add warranty for end table
Change modernscroll into end table

**Comment by kbower** [comment]
_**kbower** trac comment on Feb 4, 2011, 9:08:44 PM:_

(Changeset [[1173]](../commit/bb64a908e756563be4a2003418c8c19998beb6a7))
 * When changing products, value_products autoupdates must be invoked (APIError: Line 1 has been deleted or never existed) (fixes #653)
   * Also, when saving, process_value_lines will not set lineid (since this should happen later)
 * optimization: make attach_all_products actually helpful (fixes [#654](654))

**Comment by kbower** [comment]
_**kbower** trac status change on Feb 4, 2011, 9:08:44 PM:_ **closed**
